### PR TITLE
cmd(engine): add pkg/cmd package, use tiflow as binary name

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve?

Issue Number: ref #6022

### What is changed and how it works?

- [x] 1. Move df-executor, df-master to `tiflow executor` and `tiflow master`
- [ ] 2. refactor server master config and executor config
- [ ] 3. add more unit tests
- [ ] 4. migrate df-master-client to `tiflow cli`

Step-2 and step-3 will be added in another PR to make the PR not too large.

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: .github/workflows/dataflow_engine_chaos.yaml [hunk 1]
```diff
@@ -55,7 +55,7 @@ jobs:
           helm version
 
       - name: Build dataflow engine binary
-        run: make df-master df-executor df-chaos-case
+        run: make binary-tiflow df-chaos-case
```
Comment (CharlesCheung96) [inline .github/workflows/dataflow_engine_chaos.yaml:58]: Is it clearer to use `tiflow` here?

File: Makefile [hunk 2]
```diff
@@ -2,7 +2,7 @@
 .PHONY: build test check clean fmt cdc kafka_consumer coverage \
 	integration_test_build integration_test integration_test_mysql integration_test_kafka bank \
 	dm dm-master dm-worker dmctl dm-syncer dm_coverage \
-	engine df-master df-executor df-master-client df-demo df-chaos-case
+	engine binary-tiflow df-master-client df-demo df-chaos-case
```
Comment (CharlesCheung96) [inline Makefile:5]: Is it clearer to use `tiflow` here?

**Comment by CharlesCheung96** [comment]
image

File: Makefile (hunk 2 above)
Comment (amyangfei) [reply Makefile]: addressed

**Comment by amyangfei** [comment]
> image

fixed @CharlesCheung96

File: engine/servermaster/config.go [hunk 3]
```diff
@@ -154,74 +112,47 @@ func (c *Config) Toml() (string, error) {
...
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
-// configFromFile loads config from file.
-func (c *Config) configFromFile(path string) error {
+// ConfigFromFile loads config from file.
+func (c *Config) ConfigFromFile(path string) error {
```
Comment (gozssky) [inline engine/servermaster/config.go:155]: How about `func LoadConfig(path string) (*Config, error)`? I think it's easier to use.

File: engine/servermaster/config.go [hunk 4]
```diff
@@ -237,6 +168,11 @@ func (c *Config) configFromString(data string) error {
 	return checkUndecodedItems(metaData)
 }
 
+// GetDefaultMasterConfig returns a default master config
+func GetDefaultMasterConfig() *Config {
+	return defaultMasterConfig.Clone()
```
Comment (gozssky) [inline engine/servermaster/config.go:173]: The same to `defaultExecutorConfig`.

File: engine/servermaster/config_test.go [hunk 5]
```diff
@@ -22,28 +23,35 @@ import (
...
+
+	require.Equal(t, "mysql-0:3306", config.FrameMetaConf.Endpoints[0])
+	require.Equal(t, "root", config.FrameMetaConf.Auth.User)
+	require.Equal(t, "passwd", config.FrameMetaConf.Auth.Passwd)
+	require.Equal(t, "metastore:12479", config.UserMetaConf.Endpoints[0])
+}
+
+func mustWriteToTempFile(t *testing.T, content string) (filePath string) {
+	dir := t.TempDir()
+	fd, err := ioutil.TempFile(dir, "*")
```
Comment (gozssky) [inline engine/servermaster/config_test.go:51]:
```suggestion
	fd, err := os.CreateTemp(dir, "*")
```

Comment (gozssky) [reply engine/servermaster/config_test.go:51]: `io/ioutil` is deprecated.

**Review by gozssky — COMMENTED** [review]
Rest LGTM.

File: .github/workflows/dataflow_engine_chaos.yaml (hunk 1 above)
Comment (CharlesCheung96) [inline .github/workflows/dataflow_engine_chaos.yaml:58]:
```suggestion
        run: make tiflow df-chaos-case
```

File: Makefile [hunk 6]
```diff
@@ -2,7 +2,7 @@
 .PHONY: build test check clean fmt cdc kafka_consumer coverage \
 	integration_test_build integration_test integration_test_mysql integration_test_kafka bank \
 	dm dm-master dm-worker dmctl dm-syncer dm_coverage \
-	engine df-master df-executor df-master-client df-demo df-chaos-case
+	engine tiflow df-master-client df-demo df-chaos-case
```
[the remark this answers is not in the document]
Comment (amyangfei) [reply Makefile:5]: Yes, these binaries will be renamed in another PR.

File: engine/servermaster/config.go (hunk 3 above)
Comment (amyangfei) [reply engine/servermaster/config.go]: This function also merges configurations into Config, and only used in test. I have added some comments about it.

**Comment by amyangfei** [comment]
PTAL @gozssky @CharlesCheung96

**Review by gozssky — APPROVED** [review]

**Pull request merged**